### ISSUE TITLE
Slow rendering on Firefox Android

### ISSUE THREAD

**Issue by bikb** [mozilla/pdf.js#8863]
Link to PDF file (or attach file here):
https://mozilla.github.io/pdf.js/web/viewer.html

Configuration:
- Web browser and its version: latest Firefox and Chrome/Chromium/Builtin Android browser
- Operating system and its version: Android 7.1
- PDF.js version: latest
- Is an extension:

Steps to reproduce the problem:
1. open the demo with firefox -> slow rendering
2. open the demo with Chrome/Chromium/Builtin Android Browser -> fast rendering

What is the expected behavior? (add screenshot)

What went wrong? (add screenshot)
don't know
Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

I have inspected with Firefox WebIDE Javascript runtime capturing and with Chrome Javascript runtime capturing. I have attached the json files.

Please can someone have a look at the captures? They can easily imported to firefox / chrome developer tools! Perhaps we can detect a firefox bug. Perhaps its a Javascript issue with demo viewer.js files or dependencies.

[profile_firefox_chrome.zip](https://github.com/mozilla/pdf.js/files/1270890/profile_firefox_chrome.zip)

**Comment by bikb** [comment]
I have attached json files from Samsung Galaxy S7 captures.
I have loaded the viewer. Then I have started capturing and entered page 13.
[profile_firefox_chrome_page13.zip](https://github.com/mozilla/pdf.js/files/1272125/profile_firefox_chrome_page13.zip)

**Comment by bikb** [comment]
edited the posting title because I think its not a Samsung Galaxy issue but Android Firefox one
Please can anyone test on Firefox on Android  https://mozilla.github.io/pdf.js/web/viewer.html an compare to Chrome/Chromium/Builtin Browser/ other Browser please. I think its done in some minutes.

Thank You.

**Comment by bikb** [comment]
OK, I have searched the www an found this link:
[https://bugzilla.mozilla.org/show_bug.cgi?id=527386](url)
It seems it is a firefox rendering problem. I am wondering why this 8 year old problem was never assigned to a firefox developer? So we have the problem till today.
There is also a comment with this link:
[https://bug527386.bugzilla.mozilla.org/attachment.cgi?id=411110](url)
If I run the tests on Android it shows exactly the problem.

What I have found meanwhile: 
[https://github.com/mozilla/pdf.js/pull/8650](url)
These changes give a great performance improvement. What I also noticed on Firefox Android: After these changes the httprange requests are increadibly fast compared to the version without these changes.
